kernel/comps/virtio/src/device/input/device.rs

```diff
@@ -167,25 +163,35 @@ impl InputDevice {
         }
     }
 
-    /// Query a specific piece of information by `select` and `subsel`, and write
-    /// result to `out`, return the result size.
-    pub fn query_config_select(&self, select: InputConfigSelect, subsel: u8, out: &mut [u8]) -> u8 {
+    pub fn query_config_id_name(&self) -> String {
+        let size = self.select_config(InputConfigSelect::IdName, 0);
+        let mut data_ptr = field_ptr!(&self.config, VirtioInputConfig, data);
+        let mut out = Vec::with_capacity(size);
+        for _ in 0..size {
+            out.push(data_ptr.clone().cast::<u8>().read_once().unwrap());
+            data_ptr.byte_add(1);
+        }
+        String::from_utf8(out).unwrap()
```

Comment:
There are a few small improvements to the code:
```rust
        let size = self.select_config(InputConfigSelect::IdName, 0);

        let out = {
            // TODO: Add a general API to read this byte-by-byte.
            let mut out = Vec::with_capacity(size);
            let mut data_ptr = field_ptr!(&self.config, VirtioInputConfig, data).cast::<u8>();
            for _ in 0..size {
                out.push(data_ptr.read_once().unwrap());
                data_ptr.byte_add(1);
            }
            out
        }

        String::from_utf8(out).unwrap()
```
The points are:
 1. We can avoid `data_ptr.clone().cast::<u8>()` by simply doing it outside the loop.
 2. We can add some blank lines to make the three steps (get size, read data, do data conversion) clearer. _The same fact also applies to `query_config_prop_bits`._
 3. We can use the code block to make the fact that `data_ptr` won't be used later clearer. _The same fact also applies to `VirtioNetConfig::read`._

Reply:
Nice suggestions, thanks~